doc/source/configuration/wazuh.rst

```diff
@@ -0,0 +1,435 @@
+   * os-wazuh.key,  os-wazuh.pem
+* HTTP certificate for Kibana (port 5601) & Elasticsearch (port 9200)
+   * os-wazuh_http.key, os-wazuh_http.pem
+* Root CA certificate
+   * root-ca.key  root-ca.pem
+
+
+
+It is also possible to use externally generated certificates for wazuh-dashboard. root-ca.pem should contain the CA chain.
+Those certificates can be uploaded to ``etc/kayobe/ansible/wazuh/custom_certificates``,
+and will replace certificates generated by wazuh.
+Certificates should have the same name scheme as those generated by wazuh (typicaly <node-name>.pem)
+The key for the external certificate should be in PKCS#8 format
+(in its header it may have BEGIN PRIVATE KEY instead of BEGIN RSA PRIVATE KEY or BEGIN OPENSSH PRIVATE KEY).
+
+Example OpenSSL rune to convert to PKCS#8:
+
+``openssl pkcs8 -topk8 -nocrypt -in wazuh.key -out wazuh.key.pkcs8``
+
+TODO: document how to use a local certificate. Do we need to override all certificates?
+
+=======
+Deploy
+=======
+
+Deploy Wazuh manager:
+
+``kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/wazuh-manager.yml``
+
+If you are using the wazuh generated certificates,
+this will result in the creation of some certificates and keys (in case of custom certs adjust path to it).
+Encrypt the keys (and remember to commit to git):
+
+
+``ansible-vault encrypt --vault-password-file ~/vault.pass $KAYOBE_CONFIG_PATH/ansible/wazuh/certificates/certs/*.key``
+
+Verification
+==============
+
+The Wazuh portal should be accessible on port 443 of the Wazuh
+manager’s IPs (using HTTPS, with the root CA cert in ``etc/kayobe/ansible/qwazuh/certificates/root-ca.pem``).
+The first login should be as the admin user,
+with the opendistro_admin_password password in ``$KAYOBE_CONFIG_PATH/wazuh-secrets.yml``.
+This will create the necessary indices.
+
+Troubleshooting
+
+Logs are in ``/var/log/wazuh-indexer/wazuh.log``. There are also logs in the journal.
+
+============
+Wazuh agents
+============
+
+Make sure group mappings for wazuh-agent are added to the inventory ``etc/kayobe/inventory/groups``:
+
+.. code-block:: console
+
+  [wazuh-agent:children]
+  seed
+  overcloud
+
+
+  [wazuh:children]
+  wazuh-agent
+
+Wazuh agent playbook is located in ``etc/kayobe/ansible/wazuh-agent.yml``.
+
+Wazuh agent variables file is located in ``etc/kayobe/inventory/group_vars/wazuh-agent/wazuh-agent``.
+
+You may need to modify some variables, including:
+
+* wazuh_manager_address
+
+Deploy the Wazuh agents:
+
+``kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/wazuh-agent.yml``
+
+Verification
+=============
+
+The Wazuh agents should register with the Wazuh manager. This can be verified via the agents page in Wazuh Portal.
+Check CIS benchmark output in agent section.
+
+Additional resources:
+---------------------
+
+For times when you need to upgrade wazuh with elasticsearch to version with opensearch or you just need to deinstall all wazuh compotnents:
+Wazuh purge script: https://github.com/g0rgamesh/wazuh-server-purge
```

Comment:
Should we at least move that from private space?

Reply:
I can recreate that in stackhpc organization, I wasn't sure if we can put stuff willy-nilly into the org.